plutus-contract/src/Plutus/Contract/Trace/RequestHandler.hs

```diff
@@ -175,7 +176,7 @@ handleCurrentTime ::
 handleCurrentTime =
     RequestHandler $ \_ ->
         surroundDebug @Text "handleCurrentTime" $ do
-            TimeSlot.slotToPOSIXTime <$> Wallet.Effects.getClientSlot
+            TimeSlot.slotToBeginPOSIXTime def <$> Wallet.Effects.getClientSlot
```

Comment:
Similar issue as above. Should we always return the begining time of the current slot?